app/services/cache/MongoCacheClient.scala

```diff
@@ -239,8 +274,14 @@ class MongoCacheClient(appConfig: ApplicationConfig, db: () => DefaultDB, applic
         acc + (value._1 -> Json.parse(plainText.value))
       }
     })
-
-    collection.update(ordered = false).one(bsonIdQuery(cache.id), BSONDocument("$set" -> Json.toJson(rebuiltCache)), upsert = true) map handleWriteResult
+    collection.findOneAndUpdate(
+      filter= bsonIdQuery(cache.id),
+      update = Updates.combine(
+        Updates.set("id", rebuiltCache.id),
+        Updates.set("data",Codecs.toBson(rebuiltCache.data)),
+        Updates.set("lastUpdated", LocalDateTime.now(ZoneOffset.UTC))),
```

Comment:
❓ Why set fields individually, rather than do `Codecs.toBon(rebuildCache)`

Reply:
after having a play with this idea, it's a bit tricky to get the lastUpdated param in there - there will be a way, however this is pretty straight forward :)